### PR TITLE
Add tracing

### PR DESCRIPTION
**Pull request by rubdos**

### CODE REVIEW

**Comment by gferon** [comment]
Let's switch the calls to the log crate at the same time, then?

**Comment by rubdos** [comment]
Yeh, there's a reason it's a draft ^^

Don't know when I'll continue though.

**Comment by rubdos** [comment]
A cleaner trace for Envelopes would be beneficial:

```
2024-01-06 14:23:13 [INFO] decrypt; self=ServiceCipher { protocol_store: "...", csprng: "...", trust_root: "...", local_uuid: 9bad1xx, local_device_id: 1 } envelope=Envelope { r#type: Some(UnidentifiedSender), source_service_id: None, source_device: None, destination_service_id: Some("9bad"), timestamp: Some(170455xxx), content: Some([17, 10, 33, **LOADS MORE**, ...]), server_guid: Some("4d46a6"), server_timestamp: Some(17045509xxx), urgent: Some(false), story: Some(false), reporting_token: None }
```

File: libsignal-service-actix/src/push_service.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ impl AwcPushService {
         credentials_override: HttpAuthOverride,
     ) -> Result<ClientRequest, ServiceError> {
         let url = self.cfg.base_url(endpoint).join(path.as_ref())?;
-        log::debug!("HTTP request {} {}", method, url);
+        tracing::debug!("HTTP request {} {}", method, url);
```
Comment (gferon) [inline libsignal-service-actix/src/push_service.rs:50]: We can definitely merge this as is, but if you want to use the feature of `tracing` that adds support for structured logging, the syntax changes quite a bit.

```suggestion
        tracing::debug!(%url, %method, "HTTP request");
```

one can also rename fields in the logging statements, like so:

`tracing::debug!(http_url =% url, "HTTP request")`

You need to prefix the fields with `%` to use `fmt::Display` or `?` to use `fmt::Debug` when the type doesn't implement `tracing::Value`.

**Comment by gferon** [comment]
> A cleaner trace for Envelopes would be beneficial:
> 
> ```
> 2024-01-06 14:23:13 [INFO] decrypt; self=ServiceCipher { protocol_store: "...", csprng: "...", trust_root: "...", local_uuid: 9bad1xx, local_device_id: 1 } envelope=Envelope { r#type: Some(UnidentifiedSender), source_service_id: None, source_device: None, destination_service_id: Some("9bad"), timestamp: Some(170455xxx), content: Some([17, 10, 33, **LOADS MORE**, ...]), server_guid: Some("4d46a6"), server_timestamp: Some(17045509xxx), urgent: Some(false), story: Some(false), reporting_token: None }
> ```

Most definitely, let's impl `tracing::Value` for it!

File: libsignal-service-actix/src/push_service.rs (hunk 1 above)
Comment (rubdos) [reply libsignal-service-actix/src/push_service.rs:50]: ... that's quite a QoL improvement and would remove the need for #129 / #53 !

**Comment by rubdos** [comment]
> Most definitely, let's impl `tracing::Value` for it!

Turns out that is pretty difficult :'-)

One needs to implement `valuable::Valuable` (Value itself is Sealed), but Valuable is quite the involved visitor pattern.

**Comment by rubdos** [comment]
I'm pulling spans over await points, and I shouldn't be doing that in `libsignal-service-actix/src/push_service.rs`. Needs fixing.

File: libsignal-service-actix/src/push_service.rs (hunk 1 above)
Comment (rubdos) [reply libsignal-service-actix/src/push_service.rs]: I changed quite some calls here. I suggest we clean up while we go, this is looking very pretty already on the console :)

**Comment by rubdos** [comment]
> I'm pulling spans over await points, and I shouldn't be doing that in `libsignal-service-actix/src/push_service.rs`. Needs fixing.

I fixed most of these, let's get this pulled in! Looks rather pretty imo :)

**Pull request merged**